src/pooling.jl

```diff
@@ -166,7 +172,20 @@ function meanpool(x, k::NTuple{N, Integer}; pad=0, stride=k) where N
 end
 
 
-for pool in [:maxpool, :meanpool]
+"""
+    lppool(x, p::Number, k::NTuple; pad=0, stride=k)
+
+Perform Lp pool operation with `p`-norm and `window size `k` on input tensor `x`
+"""
```

Comment:
The PyTorch layer docstring would be a nice reference for this.

Reply:
I add 

> Can this explain a bit more?
> 
> * It should say that it requires `ndims(x) == length(k)+2`.
> * I think it should also say `lppool(x, 1, k) ./ prod(k) ≈ meanpool(x, k)`, and maybe `lppool(x, 2, k).^2 ./ prod(k) ≈ meanpool(x.^2, k)`.
> * What range of `p` is allowed? E.g. someone might expect `lppool(reshape(1:10.0,:,1,1), Inf, (3,))` to be maxpool, but it isn't... maybe anything which works for `norm` but not here should be an error?
> * Can it briefly say what types are allowed for `stride, pad`, or refer to fuller docs elsewhere?

I add docs for these aspects in new commit.

> The PyTorch layer docstring would be a good inspiration for this.

I wonder if it might be suitable to write super detailed docs here. Flux.jl would wrap pooling layer after all.

Reply:
Our docstrings for `meanpool` and `maxpool` are quite short. Given that LP-pooling is far less well known however, it warrants a bit more explanation.